core/types.go

```diff
@@ -0,0 +1,166 @@
+package core
+
+import (
+	"encoding/hex"
+	"fmt"
+	"regexp"
+)
+
+func parseHEX(input string, length int) ([]byte, error) {
+	pattern, err := regexp.Compile(fmt.Sprintf("[[:xdigit:]]{%d}", length))
+	if err != nil {
+		return nil, fmt.Errorf("Invalid pattern")
+	}
+
+	valid := pattern.MatchString(input)
+	if !valid {
+		return nil, fmt.Errorf("Invalid input: %s", input)
+	}
+
+	devAddr, err := hex.DecodeString(input)
+	if err != nil {
+		return nil, fmt.Errorf("Could not decode input: %s", input)
+	}
+
+	return devAddr, nil
+}
+
+// ParseAddr parses a 32-bit hex-encoded string
+func ParseAddr(input string) ([]byte, error) {
+	return parseHEX(input, 8)
+}
+
+// ParseEUI parses a 64-bit hex-encoded string
+func ParseEUI(input string) ([]byte, error) {
+	return parseHEX(input, 16)
+}
+
+// ParseKey parses a 128-bit hex-encoded string
+func ParseKey(input string) ([]byte, error) {
```

Comment:
That would make integrating these type into the rest of the codebase a lot safer, and indeed be more future proof.

Maybe we should move the types to a separate module (such as `types`) so we have a bit more space to do this?